bokehjs/src/coffee/models/glyphs/line.coffee

```diff
@@ -69,7 +69,7 @@ export class LineView extends XYGlyphView
       values = @_x
 
     for i in [0...values.length-1]
-      if values[i]<=val<=values[i+1]
+      if values[i]<=val<=values[i+1] or values[i+1]<=val<=values[i]
```

Comment:
So I'm not 100% sure on this, but it looks like the original code would miss hits on any line segment that "went backward" ? Worth validating before merging, though.